src/dataExplorer/components/SaveAsCellForm.tsx

```diff
@@ -170,21 +176,42 @@ class SaveAsCellForm extends PureComponent<Props, State> {
 
     const viewWithProps: View = {...view, name: cellName}
 
+    let succeeded = false
+    const redirectHandler = (dashboardId: string): void =>
+      history.push(`/orgs/${orgID}/dashboards/${dashboardId}`)
+
     try {
-      targetDashboardIDs.forEach(dashID => {
+      targetDashboardIDs.forEach((dashID, idx) => {
+        const toRedirect =
+          idx === targetDashboardIDs.length - 1 ? redirectHandler : undefined
         if (dashID === DashboardTemplate.id) {
-          onCreateDashboardWithView(orgID, newDashboardName, viewWithProps)
+          onCreateDashboardWithView(
+            orgID,
+            newDashboardName,
+            viewWithProps,
+            toRedirect
+          )
           return
         }
 
         const selectedDashboard = dashboards.find(d => d.id === dashID)
-        onCreateCellWithView(selectedDashboard.id, viewWithProps)
+        onCreateCellWithView(
+          selectedDashboard.id,
+          viewWithProps,
+          null,
+          toRedirect
+        )
       })
+      succeeded = true
     } catch (error) {
       console.error(error)
     } finally {
       this.resetForm()
-      dismiss()
+      if (succeeded) {
+        this.props.setTimeMachine('de', initialStateHelper())
```

Comment:
is this a navigation or view changing like event? how's it different from `dismiss()`? what's this do from a user's perspective?

Reply:
This is the call that resets the query builder state. Dismiss removes the modal. We aren't in the querybuilder at this point so there is no need to call dismiss

Reply:
@ShmuelLotman so can you remove the `succeeded` flag and move this into the `try` block and move `dismiss` into the `catch` block?

you really shouldn't have a conditional statement in a `finally` block. it kind of defeats the purpose of a `finally` statement. the `try/catch` statement kind of encodes the value of the `success` flag in its structure. if everything succeeds and you try to set the time machine and that succeeds, hooray, everything is good to go. if it fails, it'll drop into the catch block and you handle the failure case there.